src/main/java/org/openrewrite/jenkins/github/AddSecurityScanWorkflow.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.jenkins.github;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.yaml.ChangePropertyValue;
+import org.openrewrite.yaml.JsonPathMatcher;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.YamlParser;
+import org.openrewrite.yaml.tree.Yaml;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddSecurityScanWorkflow extends ScanningRecipe<AtomicBoolean> {
+    private static final String FILE_PATH = ".github/workflows/jenkins-security-scan.yml";
+    private static final String DEFAULT_WORKFLOW_PATH = "/org/openrewrite/jenkins/github/jenkins-security-scan.yml";
+    private static final JsonPathMatcher BRANCHES_KEY = new JsonPathMatcher("$.on.push.branches");
+    private static final String JAVA_VERSION_KEY = "jobs.security-scan.with.java-version";
+    private static final String BUILD_TOOL_KEY = "jobs.security-scan.with.java-cache";
+
+    @Option(displayName = "Branches",
+            description = "Run workflow on push to these branches.",
+            example = "main",
+            required = false)
+    @Nullable
+    List<String> branches;
+
+    @Option(displayName = "Java Version",
+            description = "Version of Java to set for build.",
+            example = "11",
+            required = false)
+    @Nullable
+    Integer javaVersion;
```

Comment:
In what you might have come to expect by now: we also have a [JavaVersion marker](https://github.com/openrewrite/rewrite/blob/9a2aa266c394673dfa2a2180b7d2f6747fff40b0/rewrite-java/src/main/java/org/openrewrite/java/marker/JavaVersion.java#L28-L35) that you can use to determine the Java version. You might want to account for projects that use a different version for src/main versus src/test, but other than that it should be convenient not to have to pass this parameter I think.